build-tools/cache_samples.sh

```diff
@@ -5,11 +5,27 @@
 # The downloaded samples are cached under /registry/samples in the devfile registry container
 set -eu
 
-# download_sample takes in a given sample name (e.g. nodejs-basic), and git clones its corresponding repository
+# clone_sample_repo clones a given git repository to the sampleDir. Before cloning it checks
+# wether or not a revision exists.
+# Parameters:
+# 1: gitRepository  (git repository url)
+# 2: sampleDir      (output directory of the clone command)
+# 3: revision       (the prefered revision of the git repo we want to clone)
+function clone_sample_repo() {
+  local repoUrl="$1"
+  local repoOutputDir="$2"
+  local repoRevision="$3"
+  if [[ $revision == "null" ]]; then
+    git clone $repoUrl $repoOutputDir
+  else
+    git clone --single-branch --branch $repoRevision $repoUrl $repoOutputDir
```

Comment:
This will only work for revisions where it's the branch name or tag and will fail for commit id.

```shell
➜  test git clone --single-branch --branch 605081dc84696eb31f217dec3c5be86a9ec38fc8 https://github.com/devfile/registry-support.git
Cloning into 'registry-support'...
warning: Could not find remote branch 605081dc84696eb31f217dec3c5be86a9ec38fc8 to clone.
fatal: Remote branch 605081dc84696eb31f217dec3c5be86a9ec38fc8 not found in upstream origin

➜  test git clone --single-branch --branch v1.0.1 https://github.com/devfile/registry-support.git
Cloning into 'registry-support'...
```

Reply:
Oh yes nice catch! On it!

Reply:
I've updated the PR in order to checkout to a given revision instead of using `--single-branch` and `--branch` args